### PR TITLE
Add support for discount links

### LINKED ISSUE #16: Discount links don't provide a way to pass the code through to the headless store.
When a shareable discount link evaluates … for example…

`theme.com/discount/freeshipping?redirect=/pages/contact`

The theme currently would redirect you to … 

`headless.com/pages/contact`, and there is a cookie for `discount_code` set to `freeshipping` but for the domain `theme.com`.

This is because the client side code isn't executed until the routing is complete — one way to get around this would be to add a search param called `discount_code` and set it to the value passed above.

e.g. 
```js
      const discount_code = getCookie('discount_code');

      function getCookie(name) {
        name = name + '=';
        var decodedCookie = decodeURIComponent(document.cookie);
        var cookies = decodedCookie.split(';');
        for (var i = 0; i < cookies.length; i++) {
          var cookie = cookies[i].trim();
          if (cookie.indexOf(name) == 0) {
            return cookie.substring(name.length, cookie.length);
          }
        }
      }

      const url = new URL(window.location.href)

      let params = new URLSearchParams(url.search);
      
      params.append("discount_code", discount_code);

      // Continue with existing logic…
```

### PR DESCRIPTION
**Pull request by benjaminsehl**
Fixes #16

### CODE REVIEW

**Comment by benjaminsehl** [comment]
Note this doesn't actually magically add discount applications to your headless site, you'd still need to grab the search param, and then save it as a cookie (or similar) and then apply the discount when you create the cart.

File: theme/layout/theme.liquid [hunk 1]
```diff
@@ -155,6 +155,27 @@
...
+          }
+        }
+      }
+
+      var url = new URL(window.location.href)
+
+      var params = new URLSearchParams(url.search);
+      
+      params.append("discount_code", discount_code);
+
```
Comment (leeuwis) [inline theme/layout/theme.liquid:178]: Seems like this part is not being used, can be removed:

```suggestion    
```

**Review by leeuwis — CHANGES_REQUESTED** [review]
@benjaminsehl thank you for taking the time to create the issue and the PR 💯 

I found one part in the code that seems redundant, after that's removed it can be merged

> Note this doesn't actually magically add discount applications to your headless site, you'd still need to grab the search param, and then save it as a cookie (or similar) and then apply the discount when you create the cart.

Yes, makes sense. That's applicable for most logic that happens in this theme 👌

**Comment by benjaminsehl** [comment]
> I found one part in the code that seems redundant, after that's removed it can be merged

Ah yes good catch! Fixed

**Review by leeuwis — APPROVED** [review]

**Pull request merged**